rdbms/src/main/java/org/spine3/server/storage/jdbc/entity/query/InsertEntityRecordsBulkQuery.java

```diff
@@ -0,0 +1,177 @@
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.lang.String.format;
+import static org.spine3.server.storage.VisibilityField.archived;
+import static org.spine3.server.storage.VisibilityField.deleted;
+import static org.spine3.server.storage.jdbc.Sql.BuildingBlock.BRACKET_CLOSE;
+import static org.spine3.server.storage.jdbc.Sql.BuildingBlock.BRACKET_OPEN;
+import static org.spine3.server.storage.jdbc.Sql.BuildingBlock.COMMA;
+import static org.spine3.server.storage.jdbc.Sql.BuildingBlock.SEMICOLON;
+import static org.spine3.server.storage.jdbc.Sql.Query.INSERT_INTO;
+import static org.spine3.server.storage.jdbc.Sql.Query.VALUES;
+import static org.spine3.server.storage.jdbc.Sql.nPlaceholders;
+import static org.spine3.server.storage.jdbc.entity.query.EntityTable.ENTITY_COL;
+import static org.spine3.server.storage.jdbc.entity.query.EntityTable.ID_COL;
+
+/**
+ * A query for {@code INSERT}-ing multiple {@linkplain EntityRecord entity records} as a bulk.
+ *
+ * @author Dmytro Dashenkov.
+ */
+public class InsertEntityRecordsBulkQuery<I> extends WriteQuery {
+
+    private static final String FORMAT_PLACEHOLDER = "%s";
+
+    private static final int COLUMNS_COUNT = 4;
+
+    private static final String SQL_TEMPLATE = INSERT_INTO + FORMAT_PLACEHOLDER +
+                                               BRACKET_OPEN + ID_COL + COMMA + ENTITY_COL + COMMA +
+                                               archived + COMMA + deleted + BRACKET_CLOSE +
+                                               VALUES + FORMAT_PLACEHOLDER;
+    private static final String SQL_VALUES_TEMPLATE = nPlaceholders(COLUMNS_COUNT);
+
+    private final Map<I, EntityRecord> records;
+    private final IdColumn<I> idColumn;
+
+    protected InsertEntityRecordsBulkQuery(Builder<I> builder) {
+        super(builder);
+        this.records = builder.records;
+        this.idColumn = builder.idColumn;
+    }
+
+    public static <I> Builder<I> newBuilder() {
+        return new Builder<>();
+    }
+
+    @Override
+    protected PreparedStatement prepareStatement(ConnectionWrapper connection) {
+        final PreparedStatement statement = super.prepareStatement(connection);
+        int parameterCounter = 1;
+        for (Map.Entry<I, EntityRecord> record : records.entrySet()) {
+            final I id = record.getKey();
+            final EntityRecord storageRecord = record.getValue();
+            addRecordsParams(statement, parameterCounter, id, storageRecord);
+            parameterCounter += COLUMNS_COUNT;
+        }
+        return statement;
+    }
+
+    /**
+     * Adds an {@link EntityRecord} to the query.
+     *
+     * <p>This includes following items:
+     * <ul>
+     *     <li>ID;
+     *     <li>Serialized record by itself;
```

Comment:
If you use a semicolon to delimit the list items, then they should not capitalize.
However, `ID` and `Boolean` are fine, as they are always capitalized. So the only issue is this line.